README.Rmd

```diff
@@ -35,62 +35,75 @@ knitr::opts_chunk$set(
 devtools::install_github("baumer-lab/fec16")
 ```
 
-```{r}
+```{r message=FALSE, warning=FALSE}
 library(fec16)
 ```
 
 
-## Data
+## Datasets Included
+
+### Full Datasets
+
+- `campaigns`: the house/senate current campaigns
+- `results_house`: the house results of the 2016 general presidential election
+- `results_senate`: the senate results of the 2016 general presidential election
+- `results_president`: the final results of the 2016 general presidential election
+
+### Sample Datasets (with 1000 rows each)
+
+- `candidates`: candidates registered with the FEC during the 2015-2016 election cycle
+- `committees`: committees registered with the FEC during the 2015-2016 election cycle
+- `individuals`: individual contributions to candidates/committees during the 2016 general presidential election
+- `contributions`: candidates and their contributions from committees during the 2016 general election
+- `expenditures`: the operating expenditures 
+- `transactions`: transactions between committees
+- `pac`: Political Action Committee (PAC) and party summary financial information
 
-- `candidates`: all candidates registered with the FEC during the 2015-2016 election cycle
-- `committees`: all committees registered with the FEC during the 2015-2016 election cycle
-- `results`: the results of the 2016 general presidential election
-- `individuals`: a sample of 5000 individual contributions to candidates/committees during the primary and general 2016 elections
-- `committee_contributions`: total contributions, aggregated by candidate, from committees
+## Functions Included
+
+The following functions retrieve the entire datasets for the sampled ones listed above. The size of the raw file that is downloaded by calling each function is given for reference. All functions have an argument `n_max` which defaults to the entire dataset but the user can specify the max length of the dataset to be loaded via this argument.
+
+- `all_candidates()` ~ 272KB
+- `all_committees()` ~ 732KB
+- `all_individuals()` ~ 1.45GB
+- `all_contributions()` ~ 15.4MB
+- `all_expenditures()` ~ 52.1MB
+- `all_transactions()` ~ 79.2MB
+- `all_pac()` ~ 399KB
+
+**Note:** When these functions are run in the console, it is helpful to know the progress of the download taking place but running these functions in a R Markdown document leads to the progress bar of the download being printed repeatedly. A possible solution is to wrap the function in `invisible(utils::capture.output())` hereby suppressing the messages. 
```

Comment:
I think we can add an option called `verbose` or `progress` and have the default be a call to `interactive()`. That way, by default it will show the progress bar in the terminal but not in the rendered Markdown. And the user can override that.

Reply:
> I think we can add an option called `verbose` or `progress` and have the default be a call to `interactive()`. That way, by default it will show the progress bar in the terminal but not in the rendered Markdown. And the user can override that.

Could you give a basic example of how this would work? Would you just call `progress` in the function, where, by default, it would run `interactive()`?